CONTRIBUTING.md

```diff
@@ -0,0 +1,78 @@
+* Describe the behavior you observed after following the steps and point out what exactly is the problem with that behavior.
+* Explain which behavior you expected to see instead and why.
+* Include screenshots which show you following the described steps and clearly demonstrate the problem.
+* If the problem is related to performance or memory, include a CPU profile capture with your report.
+* If the problem wasn't triggered by a specific action, describe what you were doing before the problem happened and share more information using the guidelines below.
+
+1. Read the [documentation](./docs) which may address known issues
+2. Search GitHub Issues for similar bugs if the documentation does not cover your issue
+3. If the bug is not reported, start by [creating a new issue](https://github.com/ProjectAGI/agi/issues/new) on GitHub
+4. Provide the necessary details in order to reproduce your bug along with relevant system information
+5. Label the issue appropriately as a `bug`
+
+### Suggesting Enhancements
+This section will guide you through submitting an enhancement suggestion for AGIEF. This includes new features and minor improvements to existing functionality.
+
+#### Before Submitting An Enhancement Suggestion
+* Determine which [repository](https://github.com/ProjectAGI) the enhancement should be suggested in.
+* Perform a [cursory search](https://github.com/search?utf8=%E2%9C%93&q=+is%3Aissue+user%3Ate a new one.
+
+#### How Do I Submit A Good Enhancement Suggestion?
+Enhancement suggestions are tracked as [GitHub issues](https://github.com/ProjectAGI/agi/issues). After you've determined which repository your enhancement suggestion is related to, create an issue on that repository and provide the following information:
+
+* Use a clear and descriptive title for the issue to identify the suggestion.
+* Provide a step-by-step description of the suggested enhancement in as many details as possible.
+* Provide specific examples to demonstrate the steps. Include copy/pasteable snippets which you use in those examples, as Markdown code blocks.
+* Describe the current behavior and explain which behavior you expected to see instead and why.
+* Include screenshots which can help you demonstrate the steps or point out the part of AGIEF which the suggestion is related to.
+* Explain why this enhancement would be useful to most users.
+* List some other frameworks where this enhancement exists.
+* Specify the name and version of the operating system you're using.
+
+### Code Contributions
+To start contributing to the codebase, ensure that you have the appropriate local development environment and adhere to the [coding guidelines](https://github.com/ProjectAGI/agi/wiki/Coding-Conventions). The development environment can be setup by following the instructions in the [documentation](./docs).
+
+#### Pull Requests
+* Fork the repository and perform the necessary code changes
```

Comment:
Do you need to fork the repo? 
I guess so that you can create a branch then do a PR ?

Reply:
It depends on if they are an active contributor with access to the repository, if not then they can't directly commit (even with a new branch) to the repository.. so they'd have to fork then make a PR